src/python/py_speech_generation.cpp

```diff
@@ -0,0 +1,194 @@
+
+#include "openvino/genai/speech_generation/speech_generation_config.hpp"
+#include "openvino/genai/speech_generation/speech_generation_perf_metrics.hpp"
+#include "openvino/genai/speech_generation/text2speech_pipeline.hpp"
+#include "py_utils.hpp"
+#include "tokenizer/tokenizers_path.hpp"
+
+namespace py = pybind11;
+using ov::genai::GenerationConfig;
+using ov::genai::PerfMetrics;
+using ov::genai::SpeechGenerationConfig;
+using ov::genai::SpeechGenerationPerfMetrics;
+using ov::genai::Text2SpeechDecodedResults;
+using ov::genai::Text2SpeechPipeline;
+
+namespace pyutils = ov::genai::pybind::utils;
+
+namespace {
+
+auto speech_generation_config_docstring = R"(
+    SpeechGenerationConfig
+    
+    Speech-generation specific parameters:
+    :param pad_token_id: padding token id.
+    :type pad_token_id: int
+
+    :param reduction_factor: reduction factor.
+    :type reduction_factor: int
+
+    :param num_mel_bins: number of Mel-filterbank channels.
+    :type num_mel_bins: int
+
+    :param minlenratio: minimum ratio of output length to input text length; prevents output that's too short.
+    :type minlenratio: float
+
+    :param maxlenratio: maximum ratio of output length to input text length; prevents excessively long outputs.
+    :type minlenratio: float
+
+    :param threshold: probability threshold for stopping decoding; when output probability exceeds above this, generation will stop.
+    :type threshold: float
+)";
+
+auto speech_generation_perf_metrics_docstring = R"(
+    Structure with raw performance metrics for each generation before any statistics are calculated.
+
+    :param num_generated_samples: Returns a number of generated samples in output
+    :type num_generated_samples: int
+)";
+
+auto text_to_speech_decoded_results = R"(
+    Structure with raw performance metrics for each generation before any statistics are calculated.
+
+    :param num_generated_samples: Returns a number of generated samples in output
+    :type num_generated_samples: int
+)";
+
+auto text_to_speech_generate_docstring = R"(
+    Generates speeches based on input texts
+
+    :param text: input text for which to generate speech
+    :type text: str
+
+    :param speaker_embedding: a vector representing the unique characteristics of a speaker's voice
+    :type speaker_embedding: openvino.Tensor
+
+    :param properties: speech generation parameters specified as properties
+    :type properties: dict
+
+    :returns: raw audios of the input texts spoken in the specified speaker's voice
+    :rtype: Text2SpeechDecodedResults
+)";
+
+SpeechGenerationConfig update_speech_generation_config_from_kwargs(const SpeechGenerationConfig& config,
+                                                                   const py::kwargs& kwargs) {
+    if (kwargs.empty())
+        return config;
+
+    SpeechGenerationConfig res_config;
+    if (!kwargs.empty())
+        res_config.update_generation_config(pyutils::kwargs_to_any_map(kwargs));
```

Comment:
I think you should update a copy of `config` instead of `res_config` with the default values. That will preserve the previous set_generation_config()

Reply:
Done.